consensus/src/test/kotlin/maru/consensus/NewSealedBlockHandlerMultiplexerTest.kt

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus
+
+import maru.core.ext.DataGenerators
+import maru.p2p.SealedBlockHandler
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.doThrow
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+
+class NewSealedBlockHandlerMultiplexerTest {
+  @Test
+  fun `should invoke all handlers for SealedBeaconBlock`() {
+    val sealedBlock = DataGenerators.randomSealedBeaconBlock(1u)
+    val handler1 =
+      mock<SealedBlockHandler> {
+        on { handleSealedBlock(any()) } doReturn SafeFuture.completedFuture(Unit)
+      }
+    val handler2 =
+      mock<SealedBlockHandler> {
+        on { handleSealedBlock(any()) } doReturn SafeFuture.completedFuture(Unit)
+      }
+    val multiplexer =
+      NewSealedBlockHandlerMultiplexer(
+        mapOf("h1" to handler1, "h2" to handler2),
+      )
+
+    val future = multiplexer.handleSealedBlock(sealedBlock)
+    future.join()
+
+    verify(handler1).handleSealedBlock(sealedBlock)
+    verify(handler2).handleSealedBlock(sealedBlock)
+    assertThat(future.isDone).isTrue()
+  }
+
+  @Test
+  fun `should log error if sealed handler throws`() {
```

Comment:
nit: this test is not checking whether an error is logged. Better name? Or add a check?